File: JavaOpenDataAccess/src/OpenDataAccess/Data/IQuery.java
package OpenDataAccess.Data;

import java.util.List;

public interface IQuery {
    String TableName(String value, boolean blGet);

    String WithSql(String value, boolean blGet);

    String EntityName(String value, boolean blGet);

    List<IDataParameterList> ParameterList(List<IDataParameterList> value, boolean blGet);

    IQuery Join(List<JoinStatement> joinList);

    IQuery Join(String jionSql);

    IQuery Where(String whereSql, List<IDataParameterList> parameterList);

    IQuery Where(List<WhereStatement> whereList, List<IDataParameterList> parameterList);

    IQuery Select(String fieldSql);

    IQuery Select(List<String> fieldList);

    IQuery OrderBy(String orderBySql);

    IQuery OrderBy(List<OrderByStatement> orderByList);

    IQuery GroupBy(List<String> groupByList);

    IQuery GroupBy(String groupBySql);

    IQuery SetSql(String sqlText, List<IDataParameterList> parameterList);

    String ToSql();

    String ToWhereSql();

    String ToGroupSql();

    String ToSelectSql();

    String ToOrderBySql();
}

File: Marriage/OpenDataAccessCore/OpenDataAccessCore.Data/IQuery.cs
using System;
using System.Collections.Generic;
using System.Data;

namespace OpenDataAccessCore.Data
{
    public interface IQuery
    {
        string TableName { get; set; }
        string WithSql { get; set; }
        string EntityName { get; set; }
        List<IDbDataParameter> ParameterList { get; set; }
        IQuery Join(List<JoinStatement> joinList);
        IQuery Join(string jionSql);
        IQuery Where(string whereSql, List<IDbDataParameter> parameterList = null);
        IQuery Where(List<WhereStatement> whereList, List<IDbDataParameter> parameterList = null);
        IQuery Select(string fieldSql);
        IQuery Select(List<string> fieldList);
        IQuery OrderBy(string orderBySql);
        IQuery OrderBy(List<OrderByStatement> orderByList);
        IQuery GroupBy(List<string> groupByList);
        IQuery GroupBy(string groupBySql);
        IQuery SetSql(string sqlText, List<IDbDataParameter> parameterList = null);
        string ToSql();
        string ToWhereSql();
        string ToGroupSql();
        string ToSelectSql();
        string ToOrderBySql();

        IQuery AddAppAccounId(Guid appAccountId);
    }
}
